File: rb/features.rb
# WorldTime SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module WorldTimeFeatures
  def self.make_feature(name)
    case name
    when "base"
      WorldTimeBaseFeature.new
    when "test"
      WorldTimeTestFeature.new
    else
      WorldTimeBaseFeature.new
    end
  end
end

File: py/worldtime_sdk/features.py
# WorldTime SDK feature factory

from worldtime_sdk.feature.base_feature import WorldTimeBaseFeature
from worldtime_sdk.feature.test_feature import WorldTimeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WorldTimeBaseFeature(),
        "test": lambda: WorldTimeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
